templates/entry.yaml

```
@@ -0,0 +1,133 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Parameters:
+  ProjectName:
+    Type: String
+  QSS3BucketName:
+    Type: String
+    Default: aws_quickstart
+  QSS3KeyPrefix:
+    Type: String
+    Default: quickstart-ammos-smallsat-toolkit/
+  QSS3BucketRegion:
+    Type: String
+    Default: ''
+  MainStackParams:
+    Type: String
+  RolePath:
+    Type: String
+    Default: ''
+  PermissionsBoundaryArn:
+    Type: String
+    Default: ''
+  DeployIam:
+    Type: String
+    Default: 'true'
+Conditions:
+  DeployIam: !Equals
+    - !Ref 'DeployIam'
+    - 'true'
+  RolePathProvided: !Not
+    - !Equals
+      - ''
+      - !Ref 'RolePath'
+  UsingDefaultBucket: !Equals
+    - !Ref 'QSS3BucketName'
+    - aws-quickstart
+  PermissionsBoundaryProvided: !Not
+    - !Equals
+      - ''
+      - !Ref 'PermissionsBoundaryArn'
+Resources:
+  ConfigParserRole:
+    Type: AWS::IAM::Role
+    Condition: DeployIam
+    Properties:
+      RoleName: !Sub '${ProjectName}-ConfigParser'
+      AssumeRolePolicyDocument:
+        Statement:
+          - Action: sts:AssumeRole
+            Effect: Allow
+            Principal:
+              Service: lambda.amazonaws.com
+        Version: '2012-10-17'
+      ManagedPolicyArns:
+        - !Sub 'arn:${AWS::Partition}:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole'
+      Path: !If
+        - RolePathProvided
+        - !Ref 'RolePath'
+        - !Ref 'AWS::NoValue'
+      PermissionsBoundary: !If
+        - PermissionsBoundaryProvided
+        - !Ref 'PermissionsBoundaryArn'
+        - !Ref 'AWS::NoValue'
+  ConfigParser:
+    Type: AWS::Lambda::Function
+    Properties:
+      FunctionName: !Sub '${ProjectName}-ConfigParser'
+      Handler: config_parser.handler
+      MemorySize: 256
+      Timeout: 300
+      Role: !Join
+        - ''
+        - - !Sub 'arn:${AWS::Partition}:iam::${AWS::AccountId}:role'
+          - !If
+            - RolePathProvided
+            - !Ref 'RolePath'
+            - /
+          - !Sub '${ProjectName}-ConfigParser'
```

I thought we talked about needing to use a conditional `!Ref` here to create an implicit dependency if creating the IAM Role? e.g.
```suggestion
      Role: !If
            - DeployIam
            - !Ref ConfigParserRole
            - !Sub "${ProjectName}-ConfigParser"
```